react/components/ProductSummaryBuyButton/ProductSummaryBuyButton.js

```diff
@@ -93,7 +93,7 @@ const ProductSummaryBuyButton = ({
             available={isAvailable}
             isOneClickBuy={isOneClickBuy}
             customToastURL={customToastURL}
-            shouldAddToCart={!shouldBeALink}
+            shouldAddToCart={selectedItem || !shouldBeALink}
```

Comment:
Could you move this conditional to the definition of `shouldBeALink`? I'd also rename and invert the `shouldBeALink` to `shouldAddToCart` to prevent needing to negate it with a `!`